DigitalLearningSolutions.Web.Tests/Services/GroupServiceTests/GroupsServiceSynchroniseGroupsTests.cs

```diff
@@ -359,17 +344,15 @@ public void
                 linkedToField: 1,
                 changesToRegistrationDetailsShouldChangeGroupMembership: true
             );
-            A.CallTo(() => groupsDataService.GetGroupsForCentre(A<int>._)).Returns(
-                new List<Group> { synchronisedGroup1, synchronisedGroup2 }
-            );
 
             // When
             groupsService.SynchroniseUserChangesWithGroups(
                 reusableDelegateDetails.Id,
                 reusableEditAccountDetailsData,
                 centreAnswersData,
                 UserTestHelper.GetDefaultRegistrationFieldAnswers(reusableDelegateDetails.CentreId),
-                null
+                null,
+                new List<Group> { synchronisedGroup1, synchronisedGroup2 }

```

Comment:
I think there should be a couple of new tests added:
* Call `SynchroniseUserChangesWithSynchronisedGroups` with `groupsDataService.GetGroupsForCentre` returning one group with `ChangesToRegistrationDetailsShouldChangeGroupMembership` = true, and one where it = false, where both groups fulfil the criteria required to appear in `groupsToAddDelegateTo`. Check that a call to `groupsDataService.AddDelegateToGroup` was only made for the correct group.
* Do the same for `AddNewlyRegisteredUserToAddNewRegistrantGroups`, but setting `ShouldAddNewRegistrantsToGroup` instead of `ChangesToRegistrationDetailsShouldChangeGroupMembership`.

(Note: I have suggested new names for these functions, below)